MainWindow/Main.cs

```diff
@@ -785,10 +785,10 @@ private void openSAV(byte[] input, string path)
                     getFieldsfromPKM = populateFieldsPK6;
                     getPKMfromFields = preparePK6;
                     extraBytes = PK6.ExtraBytes;
-                    TB_GameSync.Enabled = (SAV as SAV6).GameSyncID != 0;
-                    TB_GameSync.Text = (SAV as SAV6).GameSyncID.ToString("X16");
-                    TB_Secure1.Text = (SAV as SAV6).Secure1.ToString("X16");
-                    TB_Secure2.Text = (SAV as SAV6).Secure2.ToString("X16");
+                    TB_GameSync.Enabled = ((SAV6) SAV).GameSyncID != 0;
+                    TB_GameSync.Text = ((SAV6) SAV).GameSyncID.ToString("X16");
+                    TB_Secure1.Text = ((SAV6) SAV).Secure1.ToString("X16");
+                    TB_Secure2.Text = ((SAV6) SAV).Secure2.ToString("X16");
```

Comment:
Not so much a comment on your PR, but SAV casted to SAV6 should probably be put in a variable so we don't keep casting.

While it's possible that the .Net optimizer would handle this, it might look better.